tendrl/commons/message.py

```diff
@@ -0,0 +1,102 @@
+                       "CRITICAL": "critical"})
+    publishers = type("publishers", (),
+                      {"CEPH_INTEGRATION": "ceph-integration",
+                       "GLUSTER_INTEGRATION": "gluster-integration",
+                       "NODE_AGENT": "node-agent",
+                       "COMMONS": "commons",
+                       "ALERTING": "alerting",
+                       "PERFORMANCE_MONITORING": "performance-monitoring"})
+    """When message object intialize, message_id, timestamp and
+
+    caller are None by default, when from_json function call
+    message old message_id, time_stamp and caller is populated
+    """
+    def __init__(self, priority, publisher, payload,
+                 message_id=None, timestamp=None, node_id=None,
+                 caller=None):
+        if message_id is None:
+            self.message_id = str(uuid.uuid4())
+            self.timestamp = now()
+            # From which function, line and file error raised
+            caller = getframeinfo(stack()[1][0])
+            self.caller = {"filename": (caller.filename.rsplit('/', 1))[1],
+                           "line_no": caller.lineno,
+                           "function": caller.function}
+        else:
+            self.message_id = message_id
+            self.timestamp = parser.parse(timestamp)
+            self.caller = caller
+        self.priority = priority
+        self.publisher = publisher
+        self.node_id = tendrl_ns.node_context.node_id
+        self.payload = payload
+
+    @staticmethod
+    def from_json(json_str):
+        message_json = json.loads(json_str)
+        message = Message(**message_json)
+        if not message.validate():
+            # Invalid message logged as debug
+            message_new = Message(Message.priorities.DEBUG,
+                                  Message.publishers.NODE_AGENT,
+                                  {"message": message})
+            return message_new
+        else:
+            return message
+
+    @staticmethod
+    def to_json(message):
+        return json.dumps(message.__dict__, default=serialize_message)
+
+    def validate(self):
+        """Validation for the object
+
+        check all the mandatory fields are present,
+        check payload is in dict format,
+        if payload contains job id then it is considered
+        as job_updates, job updates should have flow id,
+        Check priorities and publishers are correct
+        """
+
+        # Check payload type is dict
+        if type(self.payload) != dict:
+            return False
+        # Check mandatory fields
+        if (self.priority not in Message.priorities.__dict__.values() or
+            self.publisher not in Message.publishers.__dict__.values() or
+            self.node_id is None or
+                "message" not in self.payload):
+            return False
+        if "request_id" in self.payload:
+            if "flow_id" not in self.payload:
```

Comment:
@nthomas-redhat here the idea is payload must have message, request id is optional, but if request id came then flow_id should come.
request is not mandatory. message is mandatory. you can give anything in message part.
for example:
     @anmolbabu wants to pass alert details using message object, so what he can do is all the alert details he can pass under message part in payload.
    for job updates payload message part contains job update details and give request_id and flow_id. so if request_id is present the message handler automatically understand it is job updates.